### PR TITLE
Remove global time override in storybook config

### LINKED ISSUE #5617: Time Machine is not behaving correctly in Simorgh storybook.
**Describe the bug**
We expect storybook stories to use the actual time if we don't use the WithTimeMachine higher order component, and use the time we specify if we do. Currently, if we don't use WithTimeMachine, the time is set to a time that had been set in a different story, and if we do use it the time is reset to the actual time.

**To Reproduce**
Steps to reproduce the behaviour:

1. Go to a time dependent story in simorgh storybook that doesn't use time machine.
2. See that the time has been set to the time from a different story. 

**Expected behaviour**
Be able to set the time for each story, and don't leak this to other stories.

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Testing notes**

- [ ] This bug fix is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by EinsteinNjoroge**
Resolves #5617 

**Overall change:** 
_remove global time override in storybook config._

_Time Machine can be activated by using `WithTimeMachine` HOC on the desired components._
_When using`WithTimeMachine` HOC, `timemachine` will be reset once the component is rendered._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by DenisHdz — DISMISSED** [review]

**Review by hotinglok — CHANGES_REQUESTED** [review]
This change breaks storybook knobs. This issue was dealt with this way by adding in `startTimeMachine()` a few months ago in this PR: https://github.com/bbc/simorgh/pull/4189
Perhaps we may need to add `startTimeMachine()` somewhere else.

**Comment by DenisHdz** [comment]
> This change breaks storybook knobs. This issue was dealt with this way by adding in `startTimeMachine()` a few months ago in this PR: #4189
> Perhaps we may need to add `startTimeMachine()` somewhere else.

@lokh01 Is this due to being using `WithTimeMachine` in some stories and the `tick` being set to `false`?

```
const WithTimeMachine = ({ children }) => {
  // set time, but leave it ticking, so knobs work
  startTimeMachine(false);
  // setting tick to false, prevents the timemachine from ticking,
```
Also, is the comment `// set time, but leave it ticking, so knobs work` correct if we are setting it to false straight after?

**Comment by hotinglok** [comment]
@DenisHdz 
This was an old PR with some janky code so I'm not terribly sure, but I think you're right. The idea was to start ticking then stop it immediately so it doesn't break other stuff. Removing `startTimeMachine()` is fine if we swap `true` and `false` in `withTimeMachine.jsx` like:
```
const WithTimeMachine = ({ children }) => {
  startTimeMachine(true)
  useEffect(() => {
    resetTimeMachine();
    startTimeMachine(false);
    return resetTimeMachine();
  });
  return <>{children}</>;
};
```
But then it makes the knobs lag a bit...

File: src/testHelpers/withTimeMachine.jsx [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
   useEffect(() => {
-    resetTimeMachine();
-    startTimeMachine(true);
-    return resetTimeMachine();
+    function handleStateChange() {
+      setComponentToRender(children);
+    }
+    startTimeMachine();
+    handleStateChange();
+    return () => resetTimeMachine();
```
Comment (Bopchy) [inline src/testHelpers/withTimeMachine.jsx]: I think we should add a comment explaining why this block is needed. I don't quite understand what's happening here 😅

File: src/testHelpers/withTimeMachine.jsx [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
-  startTimeMachine(false);
-  // setting tick to false, prevents the timemachine from ticking,
-  // so the timestamp displayed is frozen, and there are no side effects.
+  const [componentToRender, setComponentToRender] = useState(null);
+
   useEffect(() => {
-    resetTimeMachine();
-    startTimeMachine(true);
-    return resetTimeMachine();
+    function handleStateChange() {
```
Comment (mmcculloch) [inline src/testHelpers/withTimeMachine.jsx]: Do we need this function? Instead we could just put `setComponentToRender(children);` below

File: src/testHelpers/withTimeMachine.jsx (hunk 1 above)
Comment (EinsteinNjoroge) [reply src/testHelpers/withTimeMachine.jsx]: In this block we're basically starting the time-machine, rendering the component, then resetting the time-machine.

In the previous implementation, the time-machine was not being reset after the component renders. That's what was making it to mis-behave.

**Review by Bopchy — APPROVED** [review]
Tested this out locally. Checked that stories that are not using time machine HOC are maintaining system time, while components rendered in stories using the time machine HOC have the set time

File: src/testHelpers/withTimeMachine.jsx (hunk 1 above)
Comment (Bopchy) [reply src/testHelpers/withTimeMachine.jsx]: Thanks for the explanation.

**Review by AlistairGempf — APPROVED** [review]
Good job with this.

Tested rendering a `<p>{new Date().toUTCString()}</p>` on `FrontPageMain` both with and without the time machine decorator. I also added the same paragraph to `FrontPageSection` to check that switching between a story with time machine and one without works as expected, and it does!

Without time machine: 
image

With time machine:
image

**Review by hotinglok — APPROVED** [review]

**Comment by paruchurisilpa** [comment]
I am having trouble testing this as not finding any storied with reative time stamp that can be tested with time machine and without..I have gone through this with Einstien and Alistair but couldn't complete testing. Will have to discuss and find a way of testing this tomorrow.

**Comment by paruchurisilpa** [comment]
Got this tested with help from Einstein by making changes locally. Works as expected..

Attached are the screenshots.

Front Pages with the time machine
Article page without the time machine

**Before fix:**

Screen Shot 2020-03-06 at 12 08 17 PM
Screen Shot 2020-03-06 at 12 07 59 PM

**After fix:**

Screen Shot 2020-03-06 at 12 09 31 PM
Screen Shot 2020-03-06 at 12 09 22 PM

**Pull request merged**